release_util/management/commands/show_unapplied_migrations.py

```diff
@@ -2,50 +2,88 @@
 
-    def handle(self, *args, **kwargs):
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--fail_on_unapplied',
+            dest='fail_on_unapplied',
+            action='store_true',
+            help="If flag specified, command will exit with a non-zero value when unapplied migrations exist.",
+        )
+        parser.add_argument(
+            '--database',
+            dest='database',
+            default=DEFAULT_DB_ALIAS,
+            help='Nominates a database to synchronize. Defaults to the "default" database.',
+        )
+        parser.add_argument(
+            '--output_file',
+            dest='output_file',
+            default=None,
+            help="Filename to which output should be written."
+        )
 
-        unapplied = []
+    def handle(self, *args, **kwargs):
 
+        unapplied = defaultdict(list)
         out = StringIO()
-        ansi_escape = re.compile(r'\x1b[^m]*m')
-        for db in settings.DATABASES.keys():
-
-            try:
-                call_command("showmigrations", database=db, stdout=out, format="list")
-            except OperationalError:
-                self.stderr.write("Unable to check migrations: cannot connect to database '{}'.\n".format(db))
-                self.stdout.write("null")
-                sys.exit(1)
-
-            for line in out.getvalue().split('\n'):
-                line = ansi_escape.sub('', line)
-                if not line.startswith(' '):
-                    # This is an application line.
-                    current_app = line.strip()
-                elif '[ ]' in line:
-                    unapplied.append((current_app, line.strip().replace('[ ] ', '')))
-
-        if unapplied:
-            self.stdout.write(json.dumps(unapplied))
+
+        # This implementation is mostly taken from the Django 'showmigrations' mgmt command.
+        db = kwargs['database']
+        connection = connections[db]
+        loader = MigrationLoader(connection, ignore_no_migrations=True)
+        graph = loader.graph
+        app_names = sorted(loader.migrated_apps)
+        # For each app, print its migrations in order from oldest (roots) to
+        # newest (leaves).
+        for app_name in app_names:
+            shown = set()
+            for node in graph.leaf_nodes(app_name):
+                for plan_node in graph.forwards_plan(node):
+                    if plan_node not in shown and plan_node[0] == app_name:
+                        if not plan_node in loader.applied_migrations:
+                            unapplied[app_name].append(plan_node[1])
+                        shown.add(plan_node)
+
+        # Compose the output YAML.
+        yaml_output = "migrations:\n"
+        for app, migrations in unapplied.items():
+            yaml_output += "  {}:\n".format(app)
+            for migration in sorted(migrations):
+                yaml_output += "    - {}\n".format(migration)
```

Comment:
Might it be better to run this through pyyaml as a dictionary or other data structure? it would allow us to ensure that the output being produced is correct yaml and allow us to catch any errors in the yaml produced during the runtime, not by the scripts that would then need to import this data down stream?

thoughts?

We could do something like this:
```
>>> d = {"migrations": [{'app_name': ['001', '002']}]}
>>> print yaml.dump(d)
migrations:
- app_name: ['001', '002']
```